server/models.py

```diff
@@ -600,6 +605,18 @@ def _unflag_all(self, member_ids):
         if backup:
             backup.flagged = False
 
+    @transaction
+    def publish_grades(self, tag, hide=False):
+        """Publish student grades for the assignment."""
+        published = tag in self.published_scores
+        if hide and published:
+            self.published_scores = [t for t in self.published_scores if t != tag]
+        elif not hide and not published:
+            # No current use case for correctness tag
+            if tag == 'Revision' and 'Composition' not in self.published_scores:
+                self.published_scores = self.published_scores + [tag, 'Composition']
```

Comment:
Is the purpose of this to automatically add `Composition` if revision is selected? 
It might be surprising to a user if they ever toggle revision scores. The flash will also say something like 
> Toggled Revision Scores 

but it will also have toggled composition scores. 

I think I understand the use case that this if statement is going for - but I don't think it's worth the extra communication cost to be clear to the user about what's happening. 

Maybe something that would accomplish this is simply flashing (in the route) a warning about how composition score is not visible if the user is toggling Revision but composition is not shown.